### PR TITLE
sphinx gallery for docs

### PR DESCRIPTION
**Pull request by clewis7**
starts #486

### CODE REVIEW

**Comment by clewis7** [comment]
Currently, the issue that I am running into is that in the `conf.py` I am forcing offscreen rendering but then when I go to make the html the wgpu canvas is rendering which causes the build process to stall

File: docs/source/conf.py [hunk 1]
```diff
@@ -4,6 +4,26 @@
...
+ROOT_DIR = os.path.abspath(os.path.join(__file__, "..", ".."))
+sys.path.insert(0, ROOT_DIR)
+
+# -- Sphix Gallery Hackz -----------------------------------------------------
+# When building the gallery, render offscreen and don't process
+# the event loop while parsing the example
+
+
+def _ignore_offscreen_run():
+    wgpu.gui.offscreen.run = lambda: None
```
Comment (almarklein) [inline docs/source/conf.py:21]: This bit should no longer be necessary. It's from a time when the offscreen backend still fired up asyncio (it does not do so anymore). The `WGPU_FORCE_OFFSCREEN` is the only thing needed.

**Comment by almarklein** [comment]
I happen to be working on https://github.com/pygfx/pygfx/pull/679.

File: docs/source/conf.py [hunk 2]
```diff
@@ -33,6 +54,41 @@
...
+    #     "Controller",
+    #     "Camera",
+    #     "show",
+    #     "Display",
+    #     "Group",
+    #     "Scene",
+    #     "Light",
+    # },
+    "doc_module": ("fastplotlib",),
+    "image_scrapers": ("fastplotlib",),
```
Comment (almarklein) [inline docs/source/conf.py:78]: If you put "pygfx" here instead, I *think* you can use our scraper, so you don;t need the `gallery_scraper.py`.

**Comment by almarklein** [comment]
> [...] the wgpu canvas is rendering which causes the build process to stall

Fastplotlib is eagerly trying to use glfw when I try it too, ignoring the offscreen setting. I expect that this should be fixed with #431

**Pull request closed without merging**